### PR TITLE
Feat/restrict user by status pending

### PR DESCRIPTION
**Pull request by m0azam**

### CODE REVIEW

File: packages/origin-backend-utils/src/guard/UserGuard.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export class UserGuard implements CanActivate {
+    constructor(private reflector: Reflector) {}
+
+    canActivate(context: ExecutionContext): boolean {
+        const roles = this.reflector.get<Role[]>('roles', context.getHandler());
+        if (!roles) {
+            return true;
+        }
+
+        console.log('test');
```
Comment (josipbagaric) [inline packages/origin-backend-utils/src/guard/UserGuard.ts]: This should be removed.

File: packages/origin-ui-core/src/components/AdminUsersTable.tsx [hunk 2]
```diff
@@ -47,15 +48,27 @@ export function AdminUsersTable() {
...
+                entities = await adminClient.getUsersBy(
+                    requestedFilters[2]?.selectedValue,
+                    parseInt(requestedFilters[0]?.selectedValue, 10) || 0,
+                    parseInt(requestedFilters[1]?.selectedValue, 10) || 0
+                );
+            } else {
+                entities = await adminClient.getAllUsers();
+            }
+        } catch (error) {
+            if (error.toJSON().message === 'Request failed with status code 412') {
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/AdminUsersTable.tsx]: Wouldn't it be better to compare response statuses?
e.g. `error.response.status === 412` or `error.toJSON().status` if available.

File: packages/origin-backend-utils/src/guard/UserGuard.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { IUserWithRelationsIds, Role, Status } from '@energyweb/origin-backend-core';
+import { CanActivate, ExecutionContext, HttpException, Injectable } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+
+@Injectable()
+export class UserGuard implements CanActivate {
```
Comment (josipbagaric) [inline packages/origin-backend-utils/src/guard/UserGuard.ts]: Can we rename `UserGuard` to something more meaningful?
e.g. `ActiveUserGuard`

File: packages/origin-ui-core/src/components/Organization/OrganizationInvite.tsx [hunk 4]
```diff
@@ -72,6 +74,13 @@ export function OrganizationInvite() {
 
             if (error?.response?.status === 401) {
                 showNotification('Unauthorized.', NotificationType.Error);
+            } else if (error.toJSON().message === 'Request failed with status code 412') {
```
Comment (josipbagaric) [inline packages/origin-ui-core/src/components/Organization/OrganizationInvite.tsx]: Can't we use `error.response.status === 412` like above?
Comparing strings in errors seems hacky.

**Comment by kosecki123** [comment]
Props for UserGuard, but also would rename it as suggested by @JosephBagaric

**Review by josipbagaric — APPROVED** [review]

**Pull request merged**